arch/riscv/src/lib.rs

```diff
@@ -277,6 +277,7 @@ pub extern "C" fn _start_trap() {
 /// the contract as stated above.
 #[cfg(any(doc, all(target_arch = "riscv32", target_os = "none")))]
 #[link_section = ".riscv.trap"]
+#[export_name = "_start_trap"]
```

Comment:
@bradjc Seems like removal of this was actually prompted by a comment of yours. 😄  https://github.com/tock/tock/pull/4487#discussion_r2173465673 https://github.com/tock/tock/pull/4487#discussion_r2173465997

Maybe we should add some documentation stating what this symbol is needed for?